Descent3/LoadLevel.cpp

```diff
@@ -3240,7 +3240,7 @@ void ReadTerrainSkyAndLightChunk(CFILE *fp, int version) {
 
   Terrain_sky.flags = cf_ReadInt(fp);
 
-#ifdef __LINUX__
+#if defined(__LINUX__) || defined(ANDROID)
```

Comment:
I know SDL undefs `__LINUX__` on Android, but this is an ugly solution. I'd rather prefer to have a distinct define for all *NIX platforms rather than a condition on the define

Reply:
i don't really disagree, but the problem here is that [SDL2 contains literally this](https://github.com/libsdl-org/SDL/blob/SDL2/include/SDL_platform.h#L63-L67):
```
#ifdef ANDROID
#undef __LINUX__
#endif
```

SDL3, for what it's worth, does _not_ contain this. might be worth just chasing down migration to SDL3? i'd be happy to drive that; I'm gonna have to get real familiar with the SDL innards anyway for OpenXR integration.

Reply:
Maybe we can move the SDL includes into a small wrapper header file where we` #define __LINUX__` in case of `ANDROID` after their includes and only include the wrapper in our sources.

Reply:
I thought about this, but depending on include order it could end up introducing some ODR violations. (Not that SDL's behavior isn't introducing the same risk 👎 )

@Lgt2x @pzychotic @winterheart how about this - particularly in light of Louis' "prefer to have a distinct *NIX marker" idea: This appears to be using `__LINUX__` _really_ as a _POSIX_ API indicator. What if I went through (in a separate PR) and replaced any POSIX-indicative-but-not-actually-Linux-exclusive conditional to use a `POSIX` define instead? Then we could use `__LINUX__`, `__MACOSX__` and `ANDROID` (and presumably `__IPHONE__` etc in the future) to disambiguate _specific_ OS differences, while still be able to lump all "*nix" things together.

Reply:
fine with me

Reply:
Split this particular bit out into https://github.com/DescentDevelopers/Descent3/pull/492